src/tribler/core/components/tag/db/tag_db.py

```diff
@@ -189,7 +200,7 @@ def show_suggestions_condition(torrent_tag):
             return not torrent_tag.local_operation and \
                    between(torrent_tag.score, HIDE_THRESHOLD + 1, SHOW_THRESHOLD - 1)
 
-        return self._get_tags(infohash, show_suggestions_condition)
+        return self._get_tags(infohash, show_suggestions_condition, relation=relation)
 
     def get_infohashes(self, tags: Set[str]) -> List[bytes]:
```

Comment:
I think `get_infohashes` should return infohashes only with the `HAS_TAG` type of relationship to tags. It is also possible to include the check for the `HAS_TAG` relationship to the `_show_condition` function